Modules/Microsoft365DSC/DSCResources/MSFT_EXOMigrationConfig/MSFT_EXOMigrationConfig.psm1

```diff
@@ -0,0 +1,453 @@
+        [System.String]
+        [ValidateSet('Yes')]
+        $IsSingleInstance,
+
+        [Parameter()]
+        [System.String]
+        $Identity,
+
+        [Parameter()]
+        [System.Boolean]
+        $CanSubmitNewBatch,
+
+        [Parameter()]
+        [System.Boolean]
+        $SupportsCutover,
+
+        [Parameter()]
+        [System.Boolean]
+        $IsValid,
+
+        [Parameter()]
+        [System.String]
+        $ObjectState,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $Credential,
+
+        [Parameter()]
+        [System.String]
+        $ApplicationId,
+
+        [Parameter()]
+        [System.String]
+        $TenantId,
+
+        [Parameter()]
+        [System.String]
+        $CertificateThumbprint,
+
+        [Parameter()]
+        [System.String]
+        $CertificatePath,
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $CertificatePassword,
+
+        [Parameter()]
+        [Switch]
+        $ManagedIdentity,
+
+        [Parameter()]
+        [System.String[]]
+        $AccessTokens
+    )
+    Write-Verbose -Message 'Getting EXO Migration Config'
+
+    if ($Global:CurrentModeIsExport)
+    {
+        $ConnectionMode = New-M365DSCConnection -Workload 'ExchangeOnline' `
+            -InboundParameters $PSBoundParameters `
+            -SkipModuleReload $true
+    }
+    else
+    {
+        $ConnectionMode = New-M365DSCConnection -Workload 'ExchangeOnline' `
+            -InboundParameters $PSBoundParameters
+    }
+
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
+
+    #region Telemetry
+    $ResourceName = $MyInvocation.MyCommand.ModuleName -replace 'MSFT_', ''
+    $CommandName = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    $nullReturn = $PSBoundParameters
+    $nullReturn.Ensure = 'Absent'
+
+    try
+    {
+        $MigrationConfig = Get-MigrationConfig -ErrorAction Stop
+
+        $result = @{
+            IsSingleInstance                           = 'Yes'
+            Identity                                   = $MigrationConfig.Identity
+            CanSubmitNewBatch                          = $MigrationConfig.CanSubmitNewBatch
+            SupportsCutover                            = $MigrationConfig.SupportsCutover
+            IsValid                                    = $MigrationConfig.IsValid
+            ObjectState                                = $MigrationConfig.ObjectState
+            Credential                                 = $Credential
+            Ensure                                     = 'Present'
```

Comment:
Is there a case where no migration configuration would exist on a tenant? If there is always one, we should remove the Ensure parameter. Typically resources with the IsSingleInstance parameter don't include the Ensure param as well.